exchange_calendars/calendar_helpers.py

```diff
@@ -71,3 +82,96 @@ def compute_all_minutes(
             )
     out = np.concatenate(pieces).view("datetime64[ns]")
     return out
+
+
+def parse_session(
+    calendar: ExchangeCalendar,
+    session: Session,
+    param_name: str | None = None,
+    strict: bool = True,
+) -> pd.Timestamp:
+    """Parse input intended to represent a session label.
+
+    Parameters
+    ----------
+    calendar :
+        Calendar against which to evaluate `session`.
+
+    session :
+        Input to be parsed to session label. Must be valid input to
+        pd.Timestamp and have a time component of 00:00.
+
+    param_name : optional
+        Name of a parameter receiving a session label. If passed
+        then error message will make reference to the parameter by name.
+
+    strict : default: True
+        Determines behaviour if `session` parses as UTC midnight although
+        is not a session of `calendar`.
+            True - raise NotSessionError.
+            False - return UTC midnight pd.Timestamp that does not
+            represent a session.
+
+    Returns
+    -------
+    pd.Timestamp
+        pd.Timestamp (UTC with time component of 00:00). If `strict` True
+        then return will represent a session of `calendar`.
+
+    Raises
+    ------
+    TypeError
+        If `session` is not of type pd.Timestamp | str | int | float |
+            datetime.datetime.
+
+    ValueError
+        If `session` is not an acceptable single-argument input to
+        pd.Timestamp.
+
+        If `session` time component is not 00:00.
+
+        If `session` is timezone aware and timezone is not UTC.
+
+    exchange_calendars.errors.NotSessionError
+        If `strict` True and `session` parses to a valid representation of
+        a session label although it is not a session of `calendar`.
+    """
+    try:
+        ts = pd.Timestamp(session)
+    except Exception:
```

Comment:
Is the a more specific exception you can catch here?

Reply:
The try / except statement here is to identify when a user has passed a session as an object that is not a valid single-argument input to Timestamp. There are at least three errors that Timestamp can raise in this case (ValueError, TypeError and OutOfBoundsDatetime). Given the try clause contains only the command to create a Timestamp I felt it reasonable to leave the captured Exception bare in order to cover all circumstances(?).

Separately, I've revised the except clause to include the original error within the traceback so the user has visibility as to why the input failed.